lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -1073,7 +1073,7 @@ instance {-# OVERLAPS #-} Arbitrary (EpochLength, SlotId) where
 
     arbitrary = do
         (EpochLength epochLength) <- arbitrary
-        ep <- EpochNo <$> choose (0, 1000)
+        ep <- EpochNo . fromIntegral @Word32 <$> choose (0, 1000)
```

Comment:
This transformation from `Word32` to `EpochNo` seems like a recurring enough operation that it could deserve its own function:

```hs
mkEpochNo :: Word32 -> EpochNo
```

This function is (ought to be) partial, which isn't the case of `fromIntegral` which could yield wrong result. It'd be better for both readability and safety to use such a smart constructor IMO, which would make sure to fail loudly when passed a `Word32` that is too big (very much like the CBOR decoder in another module).

Reply:
> This transformation from `Word32` to `EpochNo` seems like a recurring enough operation that it could deserve its own function:
> This function is (ought to be) partial.

Done. I've called the function `unsafeEpochNo`, to reflect the fact that it may fail if the specified value is too large.